caikit/core/data_model/streams/data_stream.py

```diff
@@ -344,10 +360,23 @@ def from_header_csv(cls, filename: str, *args, **kwargs) -> "DataStream[Dict]":
     def _from_header_csv_generator(cls, filename, *csv_args, **csv_kwargs):
         # open the csv file (closure around `filename`)
         with open(filename, mode="r", encoding="utf8") as fh:
+            yield from cls._from_header_csv_buffer_generator(
```

Comment:
woooooah, I've never seen `yield from` before!

Reply:
Neither had we!